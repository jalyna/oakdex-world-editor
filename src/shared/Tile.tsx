import * as React from 'react'
import styled from 'styled-components'

interface TileProps {
  x: number,
  y: number,
  className?: string,
  children?: React.ReactNode
}

const Tile = styled.div`
  position: absolute;
  box-sizing: border-box;
  width: 16px;
  height: 16px;
`

export default function ({ x, y, className, children }: TileProps) {
  const style = {
    top: (y * 16) + 'px',
    left: (x * 16) + 'px'
  }
  return <Tile style={style} className={className}>{children}</Tile>
}
