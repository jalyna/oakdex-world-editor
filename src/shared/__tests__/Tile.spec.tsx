import * as React from 'react'
import { mount } from 'enzyme'
import Tile from '../Tile'

describe('<Tile>', () => {
  it('renders correctly', () => {
    const wrapper = mount(
      <Tile x={2} y={3} className='foo' style={{ width: '55px' }}>
        content
      </Tile>
    )
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.childAt(0).prop('style').left).toEqual('32px')
    expect(wrapper.childAt(0).prop('style').top).toEqual('48px')
    expect(wrapper.childAt(0).prop('style').width).toEqual('55px')
  })
})
