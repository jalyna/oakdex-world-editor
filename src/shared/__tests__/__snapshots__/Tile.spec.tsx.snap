// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tile> renders correctly 1`] = `
<default_1
  className="foo"
  style={
    Object {
      "width": "55px",
    }
  }
  x={2}
  y={3}
>
  <styled.div
    className="foo"
    style={
      Object {
        "left": "32px",
        "top": "48px",
        "width": "55px",
      }
    }
  >
    <StyledComponent
      className="foo"
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "lastClassName": "epnlmi",
            "rules": Array [
              "
  position: absolute;
  box-sizing: border-box;
  width: 16px;
  height: 16px;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      style={
        Object {
          "left": "32px",
          "top": "48px",
          "width": "55px",
        }
      }
    >
      <div
        className="foo sc-bdVaJa epnlmi"
        style={
          Object {
            "left": "32px",
            "top": "48px",
            "width": "55px",
          }
        }
      >
        content
      </div>
    </StyledComponent>
  </styled.div>
</default_1>
`;
